service/controller/rolebindingtemplate/resource/rolebinding/resource.go

```diff
@@ -62,44 +62,70 @@ func (r *Resource) Name() string {
 	organizations := &security.OrganizationList{}
-	{
-		labelSelector := labels.SelectorFromSet(scopes.OrganizationSelector.MatchLabels)
-		if err := r.k8sClient.CtrlClient().List(ctx, organizations, &client.ListOptions{LabelSelector: labelSelector}); err != nil {
-			if apierrors.IsNotFound(err) {
-				r.logger.Debugf(ctx, "No namespaces in organization scope %s", labelSelector.String())
-				return []string{}, nil
-			}
-			return nil, microerror.Mask(err)
-		}
-		if len(organizations.Items) == 0 {
-			r.logger.Debugf(ctx, "No namespaces in organization scope %s", labelSelector.String())
-			return []string{}, nil
+
+	if err := r.k8sClient.CtrlClient().List(ctx, organizations, &client.ListOptions{LabelSelector: labelSelector}); err != nil {
+		if apierrors.IsNotFound(err) {
+			r.logger.Debugf(ctx, "No organizations in organization scope %s", labelSelector.String())
+			return organizations, nil
 		}
+		return nil, microerror.Mask(err)
 	}
+	if len(organizations.Items) == 0 {
+		r.logger.Debugf(ctx, "No organizations in organization scope %s", labelSelector.String())
+	}
+	return organizations, nil
+}
+
+func (r *Resource) getNamespacesFromOrganizations(ctx context.Context, organizations *security.OrganizationList) ([]string, error) {
 	namespaces := []string{}
-	{
-		for _, o := range organizations.Items {
-			// get the org namespace
-			namespaces = append(namespaces, o.Status.Namespace)
-
-			// get the cluster namespaces that belong to the org namespace
-			labelSelector, err := labels.Parse(fmt.Sprintf("%s=%s,%s", label.Organization, o.Name, label.Cluster))
-			if err != nil {
+
+	for _, o := range organizations.Items {
+		// get the org namespace
+		namespaces = append(namespaces, o.Status.Namespace)
+
+		// get the cluster namespaces that belong to the org namespace
+		labelSelector, err := labels.Parse(fmt.Sprintf("%s=%s,%s", label.Organization, o.Name, label.Cluster))
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		clusterNamespaces, err := r.k8sClient.K8sClient().CoreV1().Namespaces().List(ctx, metav1.ListOptions{LabelSelector: labelSelector.String()})
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
 				return nil, microerror.Mask(err)
 			}
-			clusterNamespaces, err := r.k8sClient.K8sClient().CoreV1().Namespaces().List(ctx, metav1.ListOptions{LabelSelector: labelSelector.String()})
-			if err != nil {
-				if !apierrors.IsNotFound(err) {
-					return nil, microerror.Mask(err)
-				}
-			}
-			for _, cns := range clusterNamespaces.Items {
-				namespaces = append(namespaces, cns.Name)
-			}
 		}
+		for _, cns := range clusterNamespaces.Items {
+			namespaces = append(namespaces, cns.Name)
+		}
+	}
+	return namespaces, nil
+}
+
+func (r *Resource) getNamespacesFromScope(ctx context.Context, scopes v1alpha1.RoleBindingTemplateScopes) ([]string, error) {
+	labelSelector, err := getLabelSelectorFromScopes(scopes)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
```

Comment:
I wonder if it makes sense to extract it or just inline..

Reply:
I extracted `getLabelSelectorFromScopes` so that I could unit test it bc trust issues. Now the test is already there so I kept it that way. :shrug: Otherwise there is not much point.